copy/progress_test.go

```diff
@@ -0,0 +1,112 @@
+package copy
+
+import (
+	"fmt"
+	"os/exec"
+	"testing"
+	"time"
+
+	"github.com/containers/image/docker/daemon"
+	"github.com/containers/image/signature"
+	"github.com/containers/image/types"
+	"github.com/docker/docker/reference"
+)
+
+// THIS TEST SHOULD BE SLOW. Read inner comments for details.
+func TestCopyProgress(t *testing.T) {
+	interval := 100 * time.Millisecond
+
+	// we shell out to avoid "client is newer than server" errors from docker
+	// which will happen when importing the client straight from github.
+	// This of course presumes docker is in the path.
+	if out, err := exec.Command("/bin/sh", "-c", "docker pull golang:latest").CombinedOutput(); err != nil {
```

Comment:
Hum, so far we’ve been able to make the tests run as non-root, and to not affect any global state. (True, for example by not having any `docker-daemon:` tests among other things.)

Can this be preserved?

AFAICS the test could be testing `copyBlobFromStream` directly, with `ic.dest.PutBlob` a mock which just copies the input to `ioutil.Discard`. That would allow submitting a specially crafted `srcStream`, perhaps one which provides exactly one byte on each call, with an exactly 10 ms delay — in turn allowing exact equality checks on the offsets, more precise tests on the timings, and it could make the test run much faster.